### ISSUE TITLE
Full screen option in IOS not works

### ISSUE THREAD

**Issue by sb-developers** [stephanrauh/ngx-extended-pdf-viewer#426]
Hi,

We're using your ngx-extended-pdf-viewer in our project, ande we detect and issue in IOS devices. Full screen buttons doesn't Works. You click it but don't do anything.
 
Version: 
- Angular 7
- NGX Extended PDF Viewer: ^1.5.2

Thanks,

**Comment by stephanrauh** [comment]
Is your project a browser app, or does it run in Cordova?

BTW, when I fix a fix, I'm going to fix it in the 4.x branch. So I recommend updating to version 4.1.0.

**Comment by stephanrauh** [comment]
@snagvenkar-cci @sb-developers I've just tested the full-screen option on my iOS devices on Safari. I've found two errors:

- The full-screen option does work, but instead of maximizing the PDF viewer, it opens the entire page in full-screen mode. It should hide everything outside the PDF viewer, but that doesn't seem to work on iOS.
- The breakpoints seem to be wrong. On an iPhone 7, the "presentation mode" button is invisible, and the fallback menu entry of the kebab menu isn't shown, either.

Is this the bug you're reporting?

**Comment by sb-developers** [comment]
Hi stephanrauh 

Is your project a browser app, or does it run in Cordova? In any IOS browser app.

Is this the bug you're reporting? Yes, we found this two bugs. Specially the first one, that in our case didn't works. When you click it, don't do anything.

Thanks,

**Comment by stephanrauh** [comment]
OK, I'm going to look for a solution. Thanks for the clarification!

**Comment by sb-developers** [comment]
Hi stephanrauh, thank yoy very much. 
We spect this solution :)
Thanks,

**Comment by dnyaneshwar-n** [comment]
I'm not able to view fullscreen mode on safari browser. Please provide solution.

**Comment by stephanrauh** [comment]
Would you mind to fill in some detail? What happens when you activate the fullscreen mode?

**Comment by stephanrauh** [comment]
Safari on Mac shows the entire web application instead of focusing on the PDF viewer in full screen mode. However, everything seems to be OK. Until now, I haven't found a way to display the PDF viewer full-screen. I start to believe Safari is just broken.

As a work-around, it should be possible to force full-screen mode using something like this:

```css
/deep/ #viewerContainer {
  position: fixed !important;
  top: 0  !important;
  left: 0  !important;
  width: 100vw  !important;
  height: 100vh  !important;
}
```

But that doesn't work, either. I'm clueless. Does anybody else have an idea what's going on?

**Comment by beeing** [comment]
Hi, just did some searching and found out that fullscreen may not be fully supported https://stackoverflow.com/a/53146001

Hope this helps.

**Comment by stephanrauh** [comment]
I've got good news for you: I've implemented a working full-screen mode on Safari and iPad. iPhone generally don't support the full-screen mode, so the presentation mode button is never shown.

Caveat: my work-around is fairly radical. It temporarily moves the PDF viewer to the top of the DOM and hides the rest. That's something Angular doesn't like. However, in my tests I seem to get away with it. By definition, full-screen mode means the Angular-powered widgets are invisible, so Angular won't try to manipulate the DOM. However, it's possible that features like `[(page)]` are broken. If you want to test this, you have to use Safari. The feature already worked for the other browsers, so the workaround is used exclusively for Safari.

**Comment by stephanrauh** [comment]
The feature has landed with version 8.1.0.
